lib/src/main/java/ca/barrenechea/widget/recyclerview/decoration/StickyHeaderDecoration.java

```diff
@@ -97,12 +113,7 @@ public View findHeaderViewUnder(float x, float y) {
     }
 
     private boolean hasHeader(int position) {
-        if (position == 0) {
-            return true;
-        }
-
-        int previous = position - 1;
-        return mAdapter.getHeaderId(position) != mAdapter.getHeaderId(previous);
```

Comment:
We determine is we need a header by checking if the header id of the current item is different from the previous item. If we simply check if the id is not `-1` then we would end up adding headers to all items, including items that should share a header.